Source/lib/pdf417/encoder/PDF417ErrorCorrection.cs

```diff
@@ -154,11 +154,25 @@ internal static class PDF417ErrorCorrection
       /// <returns>the number of codewords generated for error correction</returns>
       internal static int getErrorCorrectionCodewordCount(int errorCorrectionLevel)
       {
-         if (errorCorrectionLevel < 0 || errorCorrectionLevel > 8)
-         {
-            throw new ArgumentException("Error correction level must be between 0 and 8!");
-         }
-         return 1 << (errorCorrectionLevel + 1);
+          if (errorCorrectionLevel < 0 || errorCorrectionLevel > 8)
+          {
+              throw new ArgumentException("Error correction level must be between 0 and 8!");
+          }
+          return 1 << (errorCorrectionLevel + 1);
+      }
+
+      /// <summary>
+      /// Determines the error correction level for AUTO
+      /// </summary>
+      /// <param name="errorCorrectionLevel">The error correction level (0-9)</param>
+      /// <param name="sourceCodeWords">The number of codewords for AUTO errorCorrectionLevel</param>
+      /// <returns>the number of codewords generated for error correction</returns>
+      internal static int getErrorCorrectionLevel(int errorCorrectionLevel, int sourceCodeWords)
+      {
+            if (errorCorrectionLevel == (int)PDF417ErrorCorrectionLevel.AUTO)
+                return getRecommendedMinimumErrorCorrectionLevel(sourceCodeWords);

```

Comment:
if sourceCodeWords is 0 there a ArgumentException thrown in getRecommedMinimumErrorCorrectionLevel.
The method wasn't used before. Do you think that it is possible to prevent the exception?
Perhaps modifying getRecommendedMinimumErrorCorrectionLevel returning the lowest level if sourceCodeWords is 0.